chain-impl-mockchain/src/ledger/recovery.rs

```diff
@@ -936,6 +936,31 @@ fn unpack_payload_type<R: std::io::BufRead>(
         .map_err(|error| std::io::Error::new(std::io::ErrorKind::Other, error))
 }
 
+fn pack_committee_public_keys<W: std::io::Write>(
+    keys: &[chain_vote::MemberPublicKey],
+    codec: &mut Codec<W>,
+) -> Result<(), std::io::Error> {
+    use std::convert::TryInto;
+    codec.put_u8(keys.len().try_into().unwrap())?;
+    for k in keys {
+        codec.write_all(&k.to_bytes())?;
+    }
+    Ok(())
+}
+
+fn unpack_committee_public_keys<R: std::io::BufRead>(
+    codec: &mut Codec<R>,
+) -> Result<Vec<chain_vote::MemberPublicKey>, std::io::Error> {
+    let size = codec.get_u8()?;
+    let mut result = Vec::new();
+    for _ in 0..size {
+        let buf = codec.get_bytes(chain_vote::MEMBER_PUBLIC_KEY_BYTES_LEN)?;
+        // It should be safe to unpack here since this is only used with our custom packing too.
+        result.push(chain_vote::MemberPublicKey::from_bytes(&buf).unwrap());
```

Comment:
This will blow up if the first byte in `buf` is not 0x4. Should we trust the ledger on-disk to be correct?